nuget/Microsoft.Windows.CsWinRT.Authoring.targets

```diff
@@ -4,22 +4,97 @@ Copyright (C) Microsoft Corporation. All rights reserved.
-    <None Include="$(CsWinRTPath)runtimes\win-$(Platform)\native\WinRT.Host.dll"
-          Condition="Exists('$(CsWinRTPath)runtimes\win-$(Platform)\native\WinRT.Host.dll')" 
-          Link="%(RecursiveDir)%(FileName)%(Extension)" 
-          CopyToOutputDirectory="PreserveNewest" />
+    
+    <AllItemsFullPathWithTargetPath Condition="Exists('$(CsWinRTPath)lib\net5.0\WinRT.Host.Shim.dll')" Include="$(CsWinRTPath)lib\net5.0\WinRT.Host.Shim.dll">
+      <TargetPath>WinRT.Host.Shim.dll</TargetPath>
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </AllItemsFullPathWithTargetPath>
+
+    <AllItemsFullPathWithTargetPath Condition="Exists('$(CsWinRTPath)lib\net5.0\WinRT.Runtime.dll')" Include="$(CsWinRTPath)lib\net5.0\WinRT.Runtime.dll">
+      <TargetPath>WinRT.Runtime.dll</TargetPath>
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </AllItemsFullPathWithTargetPath>
+
+    <AllItemsFullPathWithTargetPath Condition="Exists('$(CsWinRTPath)runtimes\win-$(Platform)\native\WinRT.Host.dll')" 
+                                    Include="$(CsWinRTPath)runtimes\win-$(Platform)\native\WinRT.Host.dll">
+      <TargetPath>WinRT.Host.dll</TargetPath>
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </AllItemsFullPathWithTargetPath>
   </ItemGroup>
 
-  <Target Name="CsWinRTCopySDKRefDllToOutDir" Condition="'$(CsWinRTComponent)' == 'true'" AfterTargets="ResolveRuntimePackAssets">
+  <!-- Add the RuntimeConfig to output of project reference consumers -->
+  <Target Name="CsWinRTAuthoring_AddRuntimeDependenciesToContent" 
+          AfterTargets="GenerateBuildRuntimeConfigurationFiles" 
+          BeforeTargets="GetCopyToOutputDirectoryItems;_GetCopyToOutputDirectoryItemsFromThisProject">
     <ItemGroup>
+      <AllItemsFullPathWithTargetPath Include="$(ProjectRuntimeConfigFilePath)">
+        <TargetPath>$([System.IO.Path]::GetFileName($(ProjectRuntimeConfigFilePath)))</TargetPath>
+        <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+      </AllItemsFullPathWithTargetPath>
+
+      <AllPublishItemsFullPathWithTargetPath Include="$(ProjectRuntimeConfigFilePath)">
+        <TargetPath>$([System.IO.Path]::GetFileName($(ProjectRuntimeConfigFilePath)))</TargetPath>
+        <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+      </AllPublishItemsFullPathWithTargetPath>
+    </ItemGroup>
+  </Target>
+
+  <!-- Update the project's output to include the generated WinMD.
+ 
+       ResolveableAssembly metadata decides who we can support:
+          if it is not present, we support cpp, but not c# 
+            => solution: sdk1130 error becomes a warning) 
+
+          if it is present (as 'false'), we support c#, but not cpp 
+            => solution: cppwinrt targets need to use _ResolvedNativeProjectReferencePaths)
+  -->
+  <Target Name="GetTargetPath" Returns="@(TargetPathWithTargetPlatformMoniker)">

```

Comment:
pass winmd to c++ project reference consumers

Reply:
The tradeoff with passing the winmd to C++ consumers is project references from C# consumers raises the NETSDK1130 error. This means the BgTask scenario doesn't work, so maybe we should require the manual winmd reference for now?